### PR TITLE
client: fix decode block number as u32

### PR DESCRIPTION
**Pull request by andresilva**

### CODE REVIEW

File: core/client/db/src/utils.rs [hunk 1]
```diff
@@ -318,3 +330,19 @@ pub fn read_meta<Block>(db: &dyn KeyValueDB, col_meta: Option<u32>, col_header:
...
+mod tests {
+	use super::*;
+	use runtime_primitives::testing::{Block as RawBlock, ExtrinsicWrapper};
+	type Block = RawBlock<ExtrinsicWrapper<u32>>;
+
+	#[test]
+	fn number_index_key_doesnt_panic() {
+		let id = BlockId::<Block>::Number(72340207214430721);
+		match id {
+			BlockId::Number(n) => assert!(number_index_key(n).is_err()),
```
Comment (bkchr) [inline core/client/db/src/utils.rs]: Just use expect_err.

File: core/client/src/client.rs [hunk 2]
```diff
@@ -2633,4 +2633,14 @@ pub(crate) mod tests {
...
 	}
+
+	#[test]
+	fn get_header_by_block_number_doesnt_panic() {
+		let client = test_client::new();
+
+		// backend uses u32 for block numbers, make sure we don't panic when
+		// trying to convert
+		let id = BlockId::<Block>::Number(72340207214430721);
+		assert!(client.header(&id).is_err());
```
Comment (bkchr) [inline core/client/src/client.rs]: Same.

File: core/client/db/src/utils.rs [hunk 3]
```diff
@@ -84,25 +84,33 @@ pub type NumberIndexKey = [u8; 4];
 ///
 /// In the current database schema, this kind of key is only used for
 /// lookups into an index, NOT for storing header data or others.
-pub fn number_index_key<N: TryInto<u32>>(n: N) -> NumberIndexKey {
-	let n = n.checked_into::<u32>().unwrap();
-	[
+pub fn number_index_key<N: TryInto<u32>>(n: N) -> client::error::Result<NumberIndexKey> {
+	let n = if let Some(n) = n.checked_into::<u32>() {
```
Comment (bkchr) [inline core/client/db/src/utils.rs]: Why not use try_into here?

**Comment by svyatonik** [comment]
ref: #2766

**Pull request merged**